File: BOJ/Swift/BOJ_4948.swift
import Foundation

func getPrimeSet(range: Int) -> [Bool] {
    var resultArr: [Bool] = Array(repeating: true, count: range+1)

    // 각각의 숫자 담긴 배열 생성
    var list: [Int] = []
    var i = 2
    while i < resultArr.count {
        list.append(i)
        i += 1
    }

    var x = 2
    while x*x <= range {

        if resultArr[x] {
            var k = x * x
            while k <= range {
                resultArr[k] = false
                k += x
            }
        }
        x += 1
    }
    return resultArr
}

let primes = getPrimeSet(range: 123456*2+1)

while true {
  if let input = Int(readLine()!) {
    if input == 0 {
      break
    }

    var counter = 0
    let range = input+1..<input*2+1

    for i in range {
      if primes[i] {
        counter += 1
      }
    }

    print(counter)
  }

}

File: BOJ/Java/BOJ_4948.java
/**
 * Created by changnam on 2017. 5. 29..
 */
import java.util.*;

public class BOJ_4948 {
    public static void main(String[] args){
        Scanner scin = new Scanner(System.in);

        int primeCheck[] = getPrimeSet(123456*2+1);

        while (true){
          int n = scin.nextInt();

          if (n == 0){
            break;
          }
          int counter = 0;
          for(int i=n; i<=2*n; i++){
            if (primeCheck[i] != 0) {
              counter++;
            }
          }
          System.out.println(counter);

        }

    }

    static int[] getPrimeSet(int range) {
        int result[] = new int[range];

        for(int i=2; i<result.length; i++){
            result[i] = i;
        }

        for(int i=2; i<result.length; i++){
            if (result[i] == 0)
                continue;

            for(int k = i+i; k<result.length; k+=i) {
                result[k] = 0;
            }
        }

        return result;
    }
}
